bgpd/bgp_label.c

```diff
@@ -99,6 +99,10 @@ mpls_label_t bgp_adv_label(struct bgp_dest *dest, struct bgp_path_info *pi,
 			   PEER_FLAG_FORCE_NEXTHOP_SELF))
 		return remote_label;
 
+	if (from->sort == BGP_PEER_IBGP && to->sort == BGP_PEER_IBGP &&
+	    !CHECK_FLAG(to->af_flags[afi][safi], PEER_FLAG_NEXTHOP_SELF))
```

Comment:
What if this is a reflected route and `PEER_FLAG_FORCE_NEXTHOP_SELF` is enabled?

Reply:
lets try to address it in one single if.